modules/monitoring-supported-remote-write-authentication-settings.adoc

```diff
@@ -0,0 +1,331 @@
+        basicAuth:
+          username:
+            name: rw-basic-auth <1>
+            key: user
+          password:
+            name: rw-basic-auth <1>
+            key: password
+----
+<1> The name of the `Secret` object that contains the authentication credentials.
+
+.Sample YAML for bearer token authentication using a token file
+
+The following shows a sample bearer token configuration that points to a token file. 
+It assumes that a bearer token file exists in the location specified for `bearerTokenFile`.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://authorization.example.com/api/write"
+        bearerTokenFile: "<path_to_token_file>" <1>
+----
+<1> Path to the bearer token file.
+
+.Sample YAML for bearer token authentication using a `Secret` Object
+
+The following shows bearer token settings for a `Secret` object named `rw-bearer-auth` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rw-bearer-auth
+  namespace: openshift-monitoring
+stringData:
+  token: <authentication_token> <1>
+type: Opaque
+----
+<1> The authentication token.
+
+The following shows sample bearer token config map settings that use a `Secret` object named `rw-bearer-auth` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    enableUserWorkload: true
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://authorization.example.com/api/write"
+        authorization:
+          type: Bearer
+          credentials:
+            name: rw-bearer-auth <1>
+            key: token
+----
+<1> The name of the `Secret` object that contains the authentication credentials.
+
+.Sample YAML for OAuth 2.0 authentication
+
+The following shows sample OAuth 2.0 settings for a `Secret` object named `oauth2-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: oauth2-credentials
+  namespace: openshift-monitoring
+stringData:
+  id: <oauth2_id> <1>
+  secret: <oauth2_secret> <2>
+  token: <oauth2_authentication_token> <3>
+type: Opaque
+----
+<1> The Oauth 2.0 ID.
+<2> The OAuth 2.0 secret.
+<3> The OAuth 2.0 token.
+
+The following shows an `oauth2` remote write authentication sample configuration that uses a `Secret` object named `oauth2-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://test.example.com/api/write"
+        oauth2:
+          clientId:
+            secret:
```

Comment:
shall we mention the fields such as `clientId` can either be a secret or configmap?
same for the fields in tls config: `ca`, `cert`. 
`clientSecret` and `keySecret` must be secret though.